l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -108,7 +106,7 @@ contract ExecutorFacet is Base, IExecutor {
         bytes32 _expectedSystemContractUpgradeTxHash
     )
         internal
-        pure
+        view
```

Comment:
This change was needed because you now access the storage, right?
I see value in keeping this function pure (as it provides more insight on what it can and cannot do); maybe we'll change the code so that the rollup pricing mode is passed as an argument?

Reply:
Pure again in [a99c796](https://github.com/matter-labs/era-contracts/pull/225/commits/a99c79683c18cbda8f23b1310255d3710ffe1697)!